intellihide.js

```diff
@@ -221,39 +200,36 @@ const Intellihide = new Lang.Class({
 
             let monitorIndex = this._settings.get_int('preferred-monitor');
 
-            if (monitorIndex < 0 || monitorIndex > Main.layoutManager.monitors.length)
+            if ((monitorIndex < 0) || (monitorIndex > Main.layoutManager.monitors.length))
                 monitorIndex = Main.layoutManager.primaryIndex;
 
             let topWindow = null;
-            for (let i = windows.length-1; i>=0; i--) {
+            for (let i = windows.length - 1; i >= 0; i--) {
                 let meta_win = windows[i].get_meta_window();
-                if (this._handledWindow(meta_win)
-                    && meta_win.get_monitor() == monitorIndex) {
+                if (this._handledWindow(meta_win) && (meta_win.get_monitor() == monitorIndex)) {
                     topWindow = meta_win;
                     break;
                 }
             }
 
             if (topWindow !== null) {
-
                 this._topApp = this._tracker.get_window_app(topWindow);
                 // If there isn't a focused app, use that of the window on top
                 this._focusApp = this._tracker.focus_app || this._topApp
 
                 windows = windows.filter(this._intellihideFilterInteresting, this);
 
-                for(let i=0; i< windows.length; i++) {
-
+                for (let i in windows.length) {
```

Comment:
No problem. The reason It's taking a while to merge the changes is that I'm trying to to be sure to not introduce bugs with these huge commits, as it will be very difficult to spot the difference in the future.